In [94]:
import pandas as pd
import numpy as np
 

from matplotlib import pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier

In [56]:
data1=pd.read_csv('dtx.csv')
data1.drop(['Unnamed: 0'],axis=1,inplace=True)
data2=pd.read_csv('dty.csv')
data2.drop(['Unnamed: 0'],axis=1,inplace=True)

In [57]:
data1.head(2)

age  job  education  default  balance  personal_loan  Current_campaign
0   30   10          0        0     1475              0                 0
1   33    7          1        0     2030              1                 0

In [58]:
data2.head(2)

subscribed
0           0
1           0

In [59]:
dt=pd.concat([data1,data2],axis=1)

In [60]:
dt.head(2)

age  job  education  default  balance  personal_loan  Current_campaign  \
0   30   10          0        0     1475              0                 0   
1   33    7          1        0     2030              1                 0   

   subscribed  
0           0  
1           0

In [61]:
x=dt.iloc[:,:-1].values
y=dt['subscribed'].values
x.shape

(4521, 7)

In [98]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=0)
print(x_train[:5])

[[  40    9    1    0  330    1    0]
 [  47    7    1    0  694    0    0]
 [  52    9    1    0 1871    1    0]
 [  31    0    1    0  274    0    4]
 [  33    1    1    0  113    1    2]]


In [99]:
knn=KNeighborsClassifier(n_neighbors=5)
knn.fit(x_train,y_train)
y_pred=knn.predict(x_test)

In [100]:
ls=[y_test!=y_pred]


In [101]:
from sklearn import metrics

In [102]:
print("Accuracy is",metrics.accuracy_score(y_test,y_pred)*100, '%')

Accuracy is 86.51436993367723 %


## Applying KNN method we came to a conclusion that our Analysis is 86.5% accurate